File: core/src/com/mygdx/game/handler/BuyLandHandler.java
package com.mygdx.game.handler;

import com.mygdx.game.actors.Actor1;
import com.mygdx.game.MainGame;
import com.mygdx.game.entity.ConfirmResult;
import com.mygdx.game.entity.LandPoint;

public class BuyLandHandler extends BaseHandler {
    @Override
    public HandlerEntity doHandle(HandlerEntity s, HandlerChain chain) {

        ConfirmResult confirmResult = buyLandConfirm();
        if (confirmResult.isOk()) {
            Actor1 player = s.getPlayer();
            LandPoint landPoint = s.getPlayer().getCurrent().getLandPoint();
            player.buy(landPoint);
            buyLand(player);
        } else {

        }

//        s.setAny();
        return chain.process(s);
    }

    public ConfirmResult buyLandConfirm() {
        ResultReporter<ConfirmResult> reporter = new ResultReporter<>();
        MainGame.Instance.showConfirmWindow("will you want to buy this land ?", reporter);
        return reporter.waitReport();
    }

    /**
     * 替换地图
     *
     * @param actor1
     * @return
     */
    public Object buyLand(Actor1 actor1) {
        ResultReporter<Object> reporter = new ResultReporter<>();
        MainGame.Instance.buyLand(actor1, reporter);
        return reporter.waitReport();
    }

}

File: core/src/main/kotlin/com/rich/diy/game/handler/BuyLandHandler.kt
package com.rich.diy.game.handler

import com.rich.diy.game.Actor1
import com.rich.diy.game.entity.ConfirmResult
import com.rich.diy.game.entity.LandPoint
import com.rich.diy.game.events.BaseEvent
import com.rich.diy.game.events.LandEvent
import org.greenrobot.eventbus.EventBus


class BuyLandHandler : BaseHandler() {
    override fun doHandle(s: HandlerEntity, chain: HandlerChain): HandlerEntity? {
        val confirmResult: ConfirmResult = buyLandConfirm()
        if (confirmResult.isOk) {
            val player: Actor1? = s.player
            val landPoint: LandPoint = s.player!!.getCurrent()!!.landPoint
            player!!.buy(landPoint)
            buyLand(player)
        } else {
        }
        return chain.process(s)
    }

    fun buyLandConfirm(): ConfirmResult {
       return LandEvent<ConfirmResult>("buyLandConfirm").apply {
           msg = "will you want to buy this land?"
       }.postAndWait()
    }

    /**
     * 替换地图
     *
     * @param actor1
     * @return
     */
    fun buyLand(actor1: Actor1?): Any {
        val reporter = ResultReporter<Any>()
        return LandEvent<Any>("buyLand").postAndWait()
    }
}